packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/services/EditingContextPersistenceService.java

```diff
@@ -91,11 +98,26 @@ public void persist(ICause cause, IEditingContext editingContext) {
                                 .flatMap(Optional::stream)
                                 .collect(Collectors.toSet());
 
+                        Set<Document> alreadyPersistedDocuments = this.semanticDataSearchService.findById(semanticDataUUID)
```

Comment:
It may be quite expensive to load the entire data from scratch juste to have a set of identifiers